crypto/threads_none.c

```diff
@@ -25,19 +25,22 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
 
 int CRYPTO_THREAD_read_lock(CRYPTO_RWLOCK *lock)
 {
-    OPENSSL_assert(*(unsigned int *)lock == 1);
+    if (!ossl_assert(*(unsigned int *)lock == 1))
+        return 0;
```

Comment:
Supposedly a side note. Even though abort() is not really appropriate to use in [non-debug build of] a library, earlier I [for one?] argued that *nevertheless* there are situations when it's not inappropriate to perform controlled crash even in library code. It was counter-argued that application code has to be given the choice *unconditionally*. But let's consider this as an example. There is only one case when this assertion doesn't hold, and it's memory corruption. Is application code supposed to consider it as possibility every time it fails to obtain a lock? [I assume that application is not bothered with assumption that current OpenSSL build was configured with no-threads.]